### ISSUE TITLE
Materialize layers (preserve()) on the MatMul path to fix SQL blowup for sklearn MLPs

### ISSUE THREAD

**Issue by amol-** [posit-dev/orbital#115]
# Goal

sklearn MLP pipelines produce impractically large SQL because intermediate layer outputs are inlined into every neuron of the next layer. `GemmTranslator` (the PyTorch path) already fixes this by calling `preserve()` to materialize each layer's output, converting exponential SQL growth into linear growth. The sklearn path goes through `MatMulTranslator`/`AddTranslator` (skl2onnx emits `MatMul` + `Add`, not `Gemm`), which still inline everything.

Measured impact:

- `MLPClassifier(hidden_layer_sizes=(100,))` (the sklearn default): ~50MB SQL, ~940s generation, ~165s execution
- `MLPRegressor(hidden_layer_sizes=(32, 32))`: 1.6MB SQL / 27s vs 26KB / 0.4s for a single layer

Done criteria: the real-world pipeline below generates SQL in seconds with size comparable to an equivalent PyTorch model (a `(100,)` network should land in the hundreds-of-KB range, not tens of MB). Consider a scaling regression test with a SQL-size / generation-time budget, mirroring the existing Gemm one.

Real-world example (credit default risk scorer at sklearn defaults — currently unusable):

```python
import numpy as np
import pandas as pd
import orbital
from orbital.types import FloatColumnType
from sklearn.neural_network import MLPClassifier
from sklearn.pipeline import Pipeline
from sklearn.preprocessing import StandardScaler

rng = np.random.default_rng(0)
X = pd.DataFrame({
    "loan_amount": rng.uniform(1_000, 50_000, 1_000),
    "income": rng.uniform(20_000, 150_000, 1_000),
    "debt_ratio": rng.uniform(0, 1, 1_000),
    "credit_lines": rng.integers(1, 15, 1_000).astype(float),
})
y = (X["debt_ratio"] * X["loan_amount"] / X["income"] > 0.4).astype(int)

pipeline = Pipeline([
    ("scaler", StandardScaler()),
    # default hidden_layer_sizes=(100,) — a size any sklearn user starts with
    ("mlp", MLPClassifier(max_iter=300, random_state=0)),
])
pipeline.fit(X, y)

orbital_pipeline = orbital.parse_pipeline(
    pipeline, features={c: FloatColumnType() for c in X.columns}
)
sql = orbital.export_sql("credit_applications", orbital_pipeline, dialect="duckdb")
# ~50MB of SQL after ~15 minutes — not executable in practice
```

# Why

The default `MLPClassifier`/`MLPRegressor` configuration is the first thing every sklearn user tries. Fifteen-minute generation and 50MB statements make sklearn NN support effectively unusable beyond toy layer sizes, blocking a credible sklearn NN announcement. The mechanism (`preserve()`) already exists and is proven by tree ensembles and the Gemm path.

# References

- src/orbital/translation/steps/nn/gemm.py (preserve() usage around line 104 — the pattern to replicate)
- src/orbital/translation/steps/matmul.py (MatMulTranslator — where materialization is missing)
- src/orbital/translation/steps/add.py (AddTranslator — bias add on the same path)
- tests/test_pipeline_steps.py (existing Gemm scaling regression test with SQL-size budget)
- TODO_NN.txt items 2/4 (same problem, fixed for Gemm only)

**Comment by amol-** [comment]
Additional measurements while implementing #116 narrow down where the blowup comes from: it is specific to the **classifier head**, not to layer count or activation choice.

All four models below are `StandardScaler` → MLP with `hidden_layer_sizes=(12,)`, 3 input features, exported to duckdb. "activation calls" counts the activation function occurrences in the generated SQL:

| Model | SQL size | activation calls |
|---|---|---|
| `MLPRegressor(activation="relu")` | 8 KB | 12 |
| `MLPRegressor(activation="tanh")` | 4 KB | 12 |
| `MLPClassifier(activation="relu")` | 5.2 MB | 7,648 |
| `MLPClassifier(activation="tanh")` | 2.6 MB | 7,488 |

The regressor emits exactly one activation per neuron, which is optimal. The classifier emits roughly 7,500 for the same 12 neurons, a ~624x duplication, from a single hidden layer. That suggests the missing materialization is not only between hidden layers: something in the classifier output head (the `ZipMap` / probability-and-label branches re-consuming the same hidden-layer expression) is duplicating the whole upstream network per output. Worth confirming whether a `preserve()` on the last hidden layer output, before the classifier head fans out, removes most of the duplication independently of the inter-layer fix.

Two side notes:

- Activation choice affects only the constant factor, not the growth: `TANH(x)` is more compact than relu's `CASE`/`GREATEST` expansion, which is why the tanh variants are about half the size. It does not change the underlying duplication.
- The regressor numbers indicate the `MatMul`/`Add` path is not unconditionally broken, so a fix targeted at the classifier head may recover most of the practical benefit.

Reproduction used `random_state=0`, `max_iter=400`, and 200 synthetic rows; sizes are stable across runs.
